app/Export/GetImages.hs

```diff
@@ -1,25 +1,161 @@
+--                                   (lecture, tutorial) = partition isLec lecAndTut
+--                                   [lectureOfsession, tutorialOfsession] = map (filter (\x -> or ([(x !! 2 !! 0) == (head session), (x !! 2 !! 0) == 'Y']))) [lecture, tutorial]
+--                                   lecture' = map (convertLec . list2tuple) lectureOfsession
+--                                   tutorial' = map (convertTut . list2tuple) tutorialOfsession
+--                               in (lecture', tutorial')
+--                               where isLec x = (x !! 1 !! 0) == 'L'
+
+parseCourseCookie :: String -> String -> ([CourseInfo], [CourseInfo])
+parseCourseCookie s session = let lecAndTut = map (splitOn "-") $ splitOn "_" s
+                                  (lecture, tutorial) = partition isLec lecAndTut
+                                  -- get lecture and tutorial in this session
+                                  [lectureOfsession, tutorialOfsession] = map (filter (\x -> or ([(x !! 2 !! 0) == (head session), (x !! 2 !! 0) == 'Y']))) [lecture, tutorial]
+                                  lecture' = map (convertLec . list2tuple) lectureOfsession
+                                  tutorial' = map (convertTut . list2tuple) tutorialOfsession
+                              in (lecture', tutorial')
+                              where isLec x = (x !! 1 !! 0) == 'L'
+
+list2tuple :: [String] -> (String, String, String)
+list2tuple [a, b, c] = (a, b, c)
+list2tuple _ = undefined
+
+-- ("CSC148H1","L5101","S") -> CourseInfo {code = "CSC148H1", section = "LEC-0501", session = "S", time = []} convert format to match database
+convertLec :: (String, String, String) -> CourseInfo
+convertLec (code, section, session) = CourseInfo {code = code, section = take 1 section ++ "EC-" ++ drop 1 section, session = session, time = [] }
+
+-- ("CSC148H1","T0501","S") -> CourseInfo {code = "CSC148H1", section = "TUT-0501", session = "S", time = []} convert format to match database
+convertTut :: (String, String, String) -> CourseInfo
+convertTut (code, section, session) = CourseInfo {code = code, section = take 1 section ++ "UT-" ++ drop 1 section, session = session, time = [] }
+
+
+-- | Get data from database for selected courses
+getTimes :: ([CourseInfo], [CourseInfo]) -> IO ([CourseInfo], [CourseInfo])
+getTimes (lecture, tutorial) = runSqlite databasePath $ do
+    runMigration migrateAll
+    lecTimes <- mapM getLectureTime lecture  -- [(String, String, String)] -> IO ([[Time]])
+    tutTimes <- mapM getTutorialTime tutorial
+    return (lecTimes, tutTimes)
+
+
+-- | Generate scheduel in special format based on Times
+getScheduleByTime :: [CourseInfo] -> [CourseInfo] -> [[String]]
+getScheduleByTime lecTimes tutTimes = let allTimes = lecTimes ++ tutTimes
+                                          schedule = replicate 13 $ replicate 5 ""
+                                          schedule' = foldl (\acc x -> addCourseToSchedule x acc) schedule allTimes
```

Comment:
You should be able to just do `foldl addCourseToSchedule schedule allTimes`, perhaps if you flip the order of arguments.